back/src/Guide/Api/Types.hs

```diff
@@ -261,19 +284,32 @@ data TraitType = Pro | Con
 instance ToSchema TraitType where
     declareNamedSchema = genericDeclareNamedSchema schemaOptions
 
-instance ToParamSchema TraitType where
-    toParamSchema _ = mempty
-        & S.type_ .~ SwaggerString
-        & S.format ?~ "Trait type"
+instance A.ToJSON TraitType where
+  toJSON = A.genericToJSON jsonOptions
+
+instance A.FromJSON TraitType where
+  parseJSON = A.genericParseJSON jsonOptions
+
+-- | Direction (Up/Down) for item or trait and their instances.
+data Direction = DirectionUp | DirectionDown
+    deriving (Eq, Show, Generic)
+
+instance ToSchema Direction where
+  declareNamedSchema = genericDeclareNamedSchema schemaOptions
+    { constructorTagModifier = \case
+        "DirectionUp" -> "up"
+        "DirectionDown" -> "down"
+        other -> error ("Direction schema: unknown tag " <> show other)
+    }
 
-instance ToHttpApiData TraitType where
-    toUrlPiece = toText . map toLower . show
+instance A.ToJSON Direction where
+  toJSON = A.toJSON . drop (length ("Direction" :: String)) . show
```

Comment:
Won't work, you'll get `"Up"` and `"Down"` instead of `"up"` and `"down"`.

It's easier to just have a case statement with two branches:

```haskell
instance A.ToJSON Direction where
  toJSON = \case
    DirectionUp -> "up"
    DirectionDown -> "down"
```